pkg/component/etcd/etcd/etcd.go

```diff
@@ -695,21 +694,10 @@ func (e *etcd) Deploy(ctx context.Context) error {
 
 	// etcd deployed for shoot cluster
 	if e.values.NamePrefix == "" {
-		// Add scrape config for druid metrics only if the role is 'main'.
-		if e.values.Role == v1beta1constants.ETCDRoleMain {
-			scrapeConfig := e.emptyScrapeConfig()
-			if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, e.client, scrapeConfig, func() error {
-				scrapeConfig.Labels = monitoringutils.Labels(e.prometheusLabel())
-				scrapeConfig.Spec = monitoringv1alpha1.ScrapeConfigSpec{
-					HonorTimestamps: ptr.To(false),
-					MetricsPath:     ptr.To("/federate"),
-					Params:          map[string][]string{"match[]": {`{job="` + druidServiceName + `",etcd_namespace="` + e.namespace + `"}`}},
-					StaticConfigs:   []monitoringv1alpha1.StaticConfig{{Targets: []monitoringv1alpha1.Target{"prometheus-" + cache.Label + ".garden.svc"}}},
-				}
-				return nil
-			}); err != nil {
-				return err
-			}
+		// TODO(rickardsjp, chrkl, istvanballok): Remove after v1.102
```

Comment:
Just a minor comment for the record (we don't need to change it now): the condition

```
// Add scrape config for druid metrics only if the role is 'main'.
if e.values.Role == v1beta1constants.ETCDRoleMain {
```

is removed, so in each shoot reconciliation loop, this snippet might try to delete the scrapeconfig twice:
once, in the scope of etcd-main, then also in the scope of etcd-events,
although trying to delete it once would be sufficient.

Actually in the unit tests I expected an additional delete request when there was no get or patch request, in the `e.values.Role != v1beta1constants.ETCDRoleMain` case, but the `testRole` is always `Main`.